DEVEL.md

```diff
@@ -189,3 +196,104 @@ into the `testdata` directory in this repository. Run `make test`.
 
 Assuming a successful test run, copy the required generated files to `config`.
 Build the adapter image with these changes and verify the functionality.
+
+## End-to-end walk-through
+
+The guide will walk you through the creation of the required OpenShift cluster and the deployment of
+the 3scale adapter and its' integration with a 3scale API.
+
+### Deploying OpenShift with Istio
+
+The upstream [Maistra project](https://github.com/Maistra/) contains a fork of Origin which we will use for this walk-through.
+Follow the [Maistra documentation](https://maistra.io/docs/install/) to install and Istio as required. The all-in-one setup is the
+likely the quickest to get started with. When creating the `Installation` custom resource, ensure 3scale is enabled:
+
+```yaml
+  threeScale:
+    enabled: true
+```
+
+Alternatively, you can tweak as required, and run [this script for CentOS/RHEL based installation from the ground up](https://gist.github.com/unleashed/dea133398805fb2daef391ff960d0479)
+
+### Create sample application
+
+The Istio project uses the following [bookinfo application](https://istio.io/docs/examples/bookinfo/) for demonstration and testing purposes.
+It consists of a set of microservices integrated within the service mesh.
+
+The Maistra project also ships this example app and we will use that for the walk-through. Log into the OpenShift cluster we created
+in the last step as a `cluster-admin` and run the [following script](https://gist.github.com/unleashed/5f1aaeed880474fc9eb1861f0c6444e8).
+Wait for all the Pods in the `bookinfo` namespace to become ready.
+
+### Create an API on 3scale
+
+We are assuming that an active 3scale account exists at this point. If not, [go create one](https://www.3scale.net/signup/).
+
+Create an API an Application Plan and an Application and note down the following:
```

Comment:
Link docs?
Configure your service https://access.redhat.com/documentation/en-us/red_hat_3scale/2-saas/html-single/deployment_options/index?lb_target=preview#configure_your_service
Create Application plan https://access.redhat.com/documentation/en-us/red_hat_3scale/2-saas/html/access_control/application-plans?lb_target=preview#how_to_create_an_application_plan
Create Developers/Applications https://access.redhat.com/documentation/en-us/red_hat_3scale/2-saas/html/api_bizops/add-developers?lb_target=preview

Reply:
@vramosp my only problem with this is that we are linking back from an open source project to some RH product docs. Ideally we would have docs for the open source project but this is not the case

Reply:
On second thoughts not a major deal as they are not versioned or anything. I'll add them and merge. Thanks